src/DynamoDbModel.php

```diff
@@ -86,11 +86,33 @@ protected function setupDynamoDb()
 
     public function save(array $options = [])
     {
-        if (!$this->getKey()) {
-            $this->fireModelEvent('creating');
+        $create = !$this->exists;
+        if ($create && $this->fireModelEvent('creating')  === false) {
+            return false;
         }
+        if (!$create && $this->fireModelEvent('updating') === false) {
+            return false;
+        }
+
+        if ($this->fireModelEvent('saving') === false) {
+            return false;
+        }
+
+        $saved = $this->newQuery()->save();
 
-        return $this->newQuery()->save();
+        if (!$saved) {
+            return $saved;
+        }
+
+        $this->exists = true;
+        if ($create) {
+            $this->wasRecentlyCreated = true;
+            $this->fireModelEvent('created');
+        } else {
+            $this->fireModelEvent('updated');
+        }
```

Comment:
err ... wow ... I should really finish the morning coffee BEFORE writing comments ... not sure what I saw the first time I looked at that, ... but I see it now, and am making the update.